addons/mail/static/src/core/common/message.js

```diff
@@ -401,6 +419,42 @@ export class Message extends Component {
     /** @param {HTMLElement} bodyEl */
     prepareMessageBody(bodyEl) {}
 
+    prettifyMessageLinks(fetchIfNotFound = true) {
+        /** @param {HTMLAnchorElement} el */
+        function getMsgLinkorigin(el) {
+            return `${el.protocol}//${el.host}`;
+        }
+
+        if (this.messageBody.el) {
+            const linkEls = this.messageBody.el.getElementsByTagName("a");
+            const msgIds = [];
+            for (const linkEl of linkEls) {
+                if (getMsgLinkorigin(linkEl) === getOrigin()) {
+                    const match = linkEl.pathname.match(/^\/mail\/message\/(\d+)$/);
+                    if (match) {
+                        const msg = this.store.Message.insert(parseInt(match[1]));
+                        if (fetchIfNotFound) {
+                            if (!msg.thread?.name) {
+                                msgIds.push(msg.id);
+                            }
+                            this.message.messagesToLinkify.push(msg);
+                        }
+                        const threadName = msg.thread?.name ?? _t("Unknown");
+                        const angleRight = "<i class='fa fa-angle-right ms-1' aria-hidden='true'/>";
+                        const commentIcon = "<i class='fa fa-comment' aria-hidden='true'/>";
+                        linkEl.innerHTML = `<span>${threadName} ${angleRight} ${commentIcon}</span>`;
+                        linkEl.classList.add("o_message_redirect");
+                    }
+                }
```

Comment:
`threadName` is user content, building html with string interpolation like this and `innerHTML=` is prone to XSS